stdlib/jdbc/src/main/java/org/ballerinax/jdbc/datasource/SQLDatasourceUtils.java

```diff
@@ -207,25 +207,33 @@ public static ObjectValue createSQLDBClient(MapValue<String, Object> clientEndpo
     }
 
     public static ErrorValue getSQLDatabaseError(SQLException exception, String messagePrefix) {
-        String sqlErrorMessage =
-                exception.getMessage() != null ? exception.getMessage() : Constants.DATABASE_ERROR_MESSAGE;
-        int vendorCode = exception.getErrorCode();
-        String sqlState = exception.getSQLState();
-        return getSQLDatabaseError(messagePrefix + sqlErrorMessage, vendorCode, sqlState);
+        String sqlErrorMessage;
+        int vendorCode;
+        String sqlState;
+        String errorMessage = messagePrefix;
+
+        if (exception instanceof DatabaseException) {
+            DatabaseException dbException = (DatabaseException) exception;
+            String reason =
+                    dbException.getMessage() != null ? dbException.getMessage() : Constants.DATABASE_ERROR_MESSAGE;
+            sqlErrorMessage = dbException.getSqlErrorMessage();
+            vendorCode = dbException.getSqlErrorCode();
+            sqlState = dbException.getSqlState();
+            errorMessage += reason;
+        } else {
+            sqlErrorMessage =
+                    exception.getMessage() != null ? exception.getMessage() : Constants.DATABASE_ERROR_MESSAGE;
+            vendorCode = exception.getErrorCode();
+            sqlState = exception.getSQLState();
+        }
+        errorMessage += sqlErrorMessage + ".";
+        return getSQLDatabaseError(errorMessage, vendorCode, sqlState);
```

Comment:
Now that we've changed the `DatabaseException` to extend from `SQLException`, isn't it possible to just override the existing methods for `getState`, `getErrorCode` etc in the `SQLException` and come up with a way to unify the logic in these if-else blocks as well, instead of checking `instanceof`?